Tuxedo v22.1.0.0.0 - Added SALT bankapp sample application helm charts

Diw-dev opened the pull request:
Signed-off-by: Diwakar Shetty <diwakar.shetty@oracle.com>

Diw-dev commented:
@Djelibeybi 
Hello Avi, could not figure out how to add you as a reviewer through the Github menu. Hence this message to please help me with reviewing this helm chart PR ?

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
````diff
@@ -0,0 +1,111 @@
+* Create the SSL certificates by running the following steps:
+
+```shell
+BANKAPP_SITE="bankapp.oracle.com"     # can be any website name (but not IP address) of your choice
````
Never use `oracle.com` in any examples. Always use `example.com`.

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,111 @@
+BANKAPP_SITE="bankapp.oracle.com"     # can be any website name (but not IP address) of your choice
+
+# create root certificate and private key
+openssl req -x509 -sha256 -nodes -days 3650 -newkey rsa:2048 -subj '/O=Oracle Inc/CN=does-not-matter' -keyout root-ca.private.key -out root-ca.crt
```
Remove `Oracle Inc` and replace with `Example Company`

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,111 @@
+  --set image.repository=`<tuxedo-bankapp image location in container registry>`  \
+  --set EnableIngress=true \
+  --set IngressTLSSecretName=`<SSL certificate secret created earlier, e.g. bankapp-credential>` \
+  --set IngressTLSHosts=`<$BANKAPP_SITE used earlier, e.g. bankapp.oracle.com>`  \
```
Replace `oracle.com` with `example.com`

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/templates/NOTES.txt:
```diff
@@ -0,0 +1,17 @@
+  export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}') 
+  export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+  copy root-ca.crt created earlier to the location from where the below curl command will be run.
+  export BANKAPP_SITE=<same value as what was used in one of the earlier steps. e.g. bankapp.oracle.com.>
```
Replace `oracle.com` with `example.com`

Djelibeybi replied:
This still needs to be changed.

Djelibeybi commented on tuxedo/README.md:
```diff
@@ -6,6 +6,7 @@ This repository contains the Helm charts for deploying the Oracle Tuxedo suite o
 
 * [Tuxedo Workstation SSL server](./charts/tuxedo-ws): sample Tuxedo Workstation (WS) SSL server.
 * [Tuxedo Load Balancer service](./charts/tuxedo-lb): optional LoadBalancer for the sample Tuxedo Workstation SSL server.
+* [Tuxedo SALT bankapp sample](./charts/tuxedo-bankapp): Sample Tuxedo SALT bankapp sample application.
```
Please use the correct product name as is used in the `docker-images` repo.

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,111 @@
+# Oracle Tuxedo `bankapp` Sample Application
```
Please use the same product name as the container image

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -1,10 +1,10 @@
-# Oracle Tuxedo `bankapp` Sample Application
+# Oracle Tuxedo Service Architecture Leveraging Tuxedo (SALT) `bankapp` Sample Application
 
 This Helm chart creates an Oracle Tuxedo  sample `bankapp` application with services sch as `INQUIRY`, `TRANSFER`, `DEPOSIT`, `WITHDRAWAL` and other similar services.
```
```suggestion
This Helm chart creates a sample Oracle Tuxedo `bankapp` application with multiple services including `INQUIRY`, `TRANSFER`, `DEPOSIT` and `WITHDRAWAL`.
```

Djelibeybi reviewed (changes requested):
Could you please also add a disclaimer in the `README.md` that this Helm chart should only be deployed internally and used for learning and development purposes and should not be deployed to any production or externally accessible environments?

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,117 @@
+
+## Using the Tuxedo `bankapp` Sample Application without Istio Ingress and SSL
+
+* To run the `Tuxedo bankapp sample application`, install this helm chart by running the following command:
```
```suggestion
* To run the Tuxedo `bankapp` sample application, install this Helm chart by running the following command:
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
````diff
@@ -0,0 +1,117 @@
+```
+
+The expected output to the above `INQUIRY` HTTP request is
+```json
+    "ACCOUNT_ID":   10000,
+    "FORMNAM":      "CBALANCE",
+    "SBALANCE":     "$1456.00"
+```
````
Please paste the full JSON output so that it's still valid JSON.

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,117 @@
+
+## Using the Tuxedo `bankapp` Sample Application with Istio Ingress and SSL
+
+If clients external to the Kubernetes cluster should be able to access the bankapp, then you can install Istio Ingress to obtain an external IP along with SSL support.
```
```suggestion
Istio Ingress can be installed to provide an external IP and SSL support which will allow you to connect to the `bankapp` application from clients external to the Kubernetes cluster.
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,117 @@
+
+If clients external to the Kubernetes cluster should be able to access the bankapp, then you can install Istio Ingress to obtain an external IP along with SSL support.
+
+* Install Istio `demo` profile using the [Istio Installation Guides](https://istio.io/latest/docs/setup/install/). This should result in the istio-ingressgateway service being installed, along with a valid EXTERNAL-IP value, as shown in the following sample listing:
```
```suggestion
* Install Istio `demo` profile using the [Istio installation guide](https://istio.io/latest/docs/setup/install/). The `demo` profile will install an `istio-ingressgateway` service and output an `EXTERNAL-IP` value, as shown in the following sample listing:
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
````diff
@@ -0,0 +1,117 @@
+
+```
+
+* To run the `Tuxedo bankapp sample application`, install this helm chart by running the following command:
````
```suggestion
* To run the Tuxedo `bankapp` sample application, install this Helm chart by running the following command:
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,119 @@
+
+## Prerequisites
+
+* Build a `tuxedo-bankapp` container image using the [Oracle Tuxedo bankapp image](https://github.com/oracle/docker-images/tree/main/OracleTuxedo/salt/samples/bankapp)
```
```suggestion
* Build a Tuxedo `bankapp` container image using the [Oracle Tuxedo `bankapp` image](https://github.com/oracle/docker-images/tree/main/OracleTuxedo/salt/samples/bankapp)
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,119 @@
+
+* Build a `tuxedo-bankapp` container image using the [Oracle Tuxedo bankapp image](https://github.com/oracle/docker-images/tree/main/OracleTuxedo/salt/samples/bankapp)
+
+* Push the `tuxedo-bankapp` image that you built in the preceding step to a private container registry and note down its location. This location will be used in the `helm install` step later in this procedure.
```
```suggestion
* Push the Tuxedo `bankapp` image that you built in the preceding step to a private container registry and note down its location. This location will be used in the `helm install` step later in this procedure.
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,119 @@
+  --set EnableIngress=true \
+  --set IngressTLSSecretName=`<SSL certificate secret created earlier, e.g. bankapp-credential>` \
+  --set IngressTLSHosts=`<$BANKAPP_SITE used earlier, e.g. bankapp.example.com>`  \
+  tuxedo-bankapp-install tuxedo/charts/tuxedo-bankapp
```
```suggestion
  tuxedo-bankapp-install oracle/tuxedo-bankapp
```

Djelibeybi commented on tuxedo/charts/tuxedo-bankapp/README.md:
```diff
@@ -0,0 +1,119 @@
+helm install \
+  --set imagePullSecrets=`<your-secret-file-for-container-registry-access>` \
+  --set image.repository=`<tuxedo-bankapp image location in container registry>`  \
+  tuxedo-bankapp-install tuxedo/charts/tuxedo-bankapp
```
```suggestion
  tuxedo-bankapp-install oracle/tuxedo-bankapp
```

Djelibeybi reviewed (changes requested):
Remember you also need to run `package_chart.sh` to create the tarball, update the chart repo metadata and commit the tarball and updated `index.yaml` so that `oracle/tuxedo-bankapp` works correctly.

Diw-dev commented:
> Remember you also need to run `package_chart.sh` to create the tarball, update the chart repo metadata and commit the tarball and updated `index.yaml` so that `oracle/tuxedo-bankapp` works correctly.

Added the `tuxedo-bankapp-0.1.0.tgz` and updated the `index.yaml` now. Thanks.
